internal/restconfig/restconfig.go

```diff
@@ -335,8 +345,8 @@ func (rcp *Producer) RunOnSelectedPrefixedContext(prefix string, function PerCon
 			return true, status.Error(err, "error retrieving the namespace for the configuration for prefix %s", prefix)
 		}
 
-		if !overridden && rcp.defaultNamespace != nil {
-			namespace = *rcp.defaultNamespace
+		if !overridden && rcp.prefixedDefaultNamespaces[prefix] != nil {
+			namespace = *rcp.prefixedDefaultNamespaces[prefix]
 		}
```

Comment:
This should still allow falling back to the default “global” namespace (`subctl diagnose` expects this):

```suggestion
		if !overridden {
			if rcp.prefixedDefaultNamespaces[prefix] != nil {
				namespace = *rcp.prefixedDefaultNamespaces[prefix]
			} else if rcp.defaultNamespace != nil {
				namespace = *rcp.defaultNamespace
			}
		}
```